outlines/text/masks.py

```diff
@@ -0,0 +1,71 @@
+import re
+from typing import Dict, Iterable
+
+import numpy as np
+
+__all__ = [
+    "create_char_set_mask",
+    "create_float_mask",
+    "create_int_mask",
+    "create_mask_from_regex",
+]
+
+
+def create_mask_from_regex(vocabulary: Dict[str, int], regex: str) -> np.ndarray:
+    """Create a token mask from a regex.
+
+    Parameters
+    ----------
+    vocabulary
+        A dictionary that contains a tokenizer's vocabulary as a map
+        between tokens and their ids.
+    regex
+        The regex that tokens need to respect.
+
+    """
+    program = re.compile(regex)
+
+    mask = np.zeros(len(vocabulary), dtype=np.bool_)
+    for token, token_id in vocabulary.items():
+        if program.match(token) is not None:
+            mask[token_id] = True
+
+    return mask
+
+
+def create_int_mask(vocabulary: Dict[str, int]) -> np.ndarray:
```

Comment:
Benchmark against using `isdigit()`

Reply:
More generally all `int_mask`, `float_mask` and `char_set` masks should be benchmarked against their equivalent Python implementation.